Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Miden-node Dockerfile
+
+#### Setup image builder
+FROM rust:1.75.0-bullseye AS builder
+
+# Install dependencies
+RUN apt-get update && apt-get -y upgrade
+RUN apt-get install -y gcc llvm clang bindgen pkg-config
+
+# Setup workdir
+WORKDIR /app
+COPY . miden-node
+RUN cd miden-node && make
+
+### Run Miden-Node
+FROM ubuntu:22.04
+RUN apt-get update && apt-get -y upgrade && apt-get install -y make libssl-dev libsqlite3-dev curl
```

Comment:
Here too, please document why the dependencies are needed.

Why do we need `-dev` here? The builder should be compiling everything, no?

Reply:
Also taken from the README: 

https://github.com/0xPolygonMiden/miden-node/blob/3c3ebbecbece73b330fe01cfa2b6a41c825c8db1/README.md?plain=1#L35

Reply:
Okay, so the idea of the builder final image split is to have smaller image sizes, and a smaller surface area for issues. The goal is to have the build dependencies only in the builder, and then copy only the final artifacts to the last image.

So I think we should remove this.